bool fresh_msg
string t_plus
bool got_reset
string[] error_list
